File: server/strategies/vernam.py
from strategies.cipher_interface import CipherInterface


class VernamCipher(CipherInterface):
    def decrypt(self, text: str, key: str) -> str:
        try:
            if len(key) < len(text):
                return "Hata: Anahtar metinden kısa olamaz."
            
            text = text.upper()
            key = key.upper()
            result = ""
            
            for i in range(len(text)):
                p = (ord(text[i]) - ord(key[i])) % 26
                result += chr(p + 65)
            return result
        except:
            return "Hata: Vernam deşifre işlemi başarısız."

File: lib/ciphers/vernam.dart
import 'cipher_base.dart';

class VernamCipher implements CipherBase {
  @override
  String get name => "Vernam";
  @override
  String get id => "vernam";

  @override
  String encrypt(String text, String key) {
    if (key.length < text.length) return "Hata: Anahtar metinden kısa olamaz.";

    text = text.toUpperCase();
    key = key.toUpperCase();
    StringBuffer result = StringBuffer();

    for (int i = 0; i < text.length; i++) {
      int p = text.codeUnitAt(i) - 65;
      int k = key.codeUnitAt(i) - 65;
      result.write(String.fromCharCode(((p + k) % 26) + 65));
    }
    return result.toString();
  }
}